src/common/components/Recaptcha/Recaptcha.test.tsx

```diff
@@ -0,0 +1,219 @@
+import { getByRole, render, waitFor } from '@testing-library/react'
+import userEvent from '@testing-library/user-event'
+import { Recaptcha } from './Recaptcha'
+import React from 'react'
+import { useGoogleReCaptcha } from 'react-google-recaptcha-v3'
+
+jest.mock('react-google-recaptcha-v3');
+
+(useGoogleReCaptcha as jest.Mock).mockImplementation(() => {
+  return {
+    executeRecaptcha: mockExecuteRecaptcha
+  }
+})
+
+let mockExecuteRecaptcha = jest.fn()
+
+describe('Recaptcha component', () => {
+  const $translate = {
+    instant: jest.fn()
+  };
+
+  const $log = {
+    warn: jest.fn(),
+    error: jest.fn()
+  }
+
+  beforeEach(() => {
+    $translate.instant.mockImplementation((input) => input)
+    mockExecuteRecaptcha = jest.fn()
+    mockExecuteRecaptcha.mockImplementation(() => Promise.resolve('token'))
```

Comment:
Are you trying to reset the mock before each test? If so you can do this.


```suggestion
    mockExecuteRecaptcha.mockClear()
    mockExecuteRecaptcha.mockImplementation(() => Promise.resolve('token'))
```

Reply:
That doesn't appear to work, not sure why. Also tried in `afterEach`

Reply:
Is it because you changed `mockExecuteRecaptcha` to be undefined on line 129?
```
mockExecuteRecaptcha = undefined
```